designer/client/src/components/toolbars/scenarioDetails/ScenarioLabels.tsx

```diff
@@ -0,0 +1,315 @@
+}
+
+export const ScenarioLabels: LabelsEdit = ({ readOnly }: Props) => {
+    const scenarioLabels: LabelOption[] = useSelector(getScenarioLabels).map(toLabelOption);
+    const initialLabelErrors = useSelector(getScenarioLabelsErrors).filter((error) =>
+        scenarioLabels.some((label) => toLabelValue(label) === error.label),
+    );
+    const [labelsErrors, setLabelsErrors] = useState<ScenarioLabelValidationError[]>(initialLabelErrors);
+    const [showEditor, setShowEditor] = useState(scenarioLabels.length !== 0);
+
+    const theme = useTheme();
+    const { menuOption } = selectStyled(theme);
+    const dispatch = useDispatch();
+
+    const [isFetching, setIsFetching] = useState(false);
+    const [isOpen, setIsOpen] = useState(false);
+    const [isEdited, setIsEdited] = useState(false);
+    const [options, setOptions] = useState<LabelOption[]>([]);
+
+    const [input, setInput] = useState("");
+    const [inputTyping, setInputTyping] = useState(false);
+    const [inputErrors, setInputErrors] = useState<ScenarioLabelValidationError[]>([]);
+
+    const handleAddLabelClick = () => {
+        setShowEditor(true);
+        setIsEdited(true);
+    };
+
+    const inputHelperText: () => string = () => {
+        if (inputErrors.length !== 0) {
+            return formatErrors(inputErrors.concat(labelsErrors));
+        }
+
+        if (!isOpen && labelsErrors.length !== 0) {
+            return formatErrors(labelsErrors);
+        }
+
+        return undefined;
+    };
+
+    const validateInput = useMemo(() => {
+        return debounce(async (newInput: string) => {
+            if (newInput !== "") {
+                const response = await HttpService.validateScenarioLabels([newInput]);
+
+                if (response.status === 200) {
+                    setInputErrors(response.data.validationErrors);
+                }
+            }
+
+            setInputTyping(false);
+        }, 500);
+    }, []);
+
+    const validateLabels = useMemo(() => {
+        return debounce(async (labels: LabelOption[]) => {
+            const values = labels.map(toLabelValue);
+            const response = await HttpService.validateScenarioLabels(values);
+
+            if (response.status === 200) {
+                const validationError = response.data.validationErrors;
+                setLabelsErrors(validationError);
+            }
+        }, 500);
+    }, []);
+
+    const fetchAvailableLabels = useCallback(async () => {
+        setIsFetching(true);
+        const { data } = await HttpService.fetchScenarioLabels();
+        setIsFetching(false);
+        return data.labels.map(toLabelOption);
+    }, []);
+
+    const setLabels = (labels: LabelOption[]) => {
+        const sortedLabels = labels.map(toLabelValue).sort((a, b) => a.localeCompare(b));
+        dispatch(editScenarioLabels(sortedLabels));
+    };
+
+    useEffect(() => {
+        validateLabels(scenarioLabels);
+
+        if (!isEdited && scenarioLabels.length === 0) {
```

Comment:
How this logic related to the validateLabels? maybe we can move it to another useEffect? Also maybe it's worth to use meaningful names for this conditions

Reply:
Also, I'm not sure what's the reason behind this logic, could you explain it?

Reply:
I moved them to separate effect and added proper comments